requirements.test.txt

```diff
@@ -0,0 +1,25 @@
+atomicwrites==1.2.1
+attrs==18.2.0
+Automat==0.7.0
+constantly==15.1.0
+coverage==4.5.1
+decorator==4.3.0
+greenlet==0.4.15
+hyperlink==18.0.0
+idna==2.7
+incremental==17.5.0
+more-itertools==4.3.0
+pluggy==0.7.1
+py==1.6.0
+PyHamcrest==1.9.0
+pytest==3.8.0
+pytest-cov==2.6.0
+pytest-twisted==1.8
+-e .[test]
+six==1.11.0
+tox==3.2.1
+Twisted==18.9.0rc1; platform_system != "Windows" and platform_system != "Darwin"
+Twisted[osx_platform]==18.9.0rc1; platform_system == "Darwin"
+Twisted[windows_platform]==18.9.0rc1; platform_system == "Windows"
+virtualenv==16.0.0
+zope.interface==4.5.0
```

Comment:
For what it's worth, some of these version-pins are out of date (off-hand, at least, I know Twisted 18.9.0 has been released, zope.interface 4.6.0 is out now -- there may be more..). If you're committed to pinning specific versions here, would it be worth also looking into using/enabling a service like [requires.io](https://requires.io/) or [pyup](https://pyup.io/) to help update such pins moving forward? (I don't currently use these services myself, but it might ease the burden on current/future maintainers if we're going to be pinning all the things).

Reply:
I don't really like versions changing behind our backs, it's helpful to know whether a PR change or a version change triggered it.  I did take a stab over in Julian/venvs#53 at having both pinned and continuously updating (even vcs) builds.  I'll do an update pass right now but can we leave automation or other management setup to another PR?

Reply:
Yes, I agree both that we should strive not to have dependencies change behind our backs and that it would make sense to address any such automations in a separate PR.